src/coreclr/vm/jitinterface.cpp

```diff
@@ -11915,10 +12000,17 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
             if (fieldObj != NULL)
             {
                 Object* obj = OBJECTREFToObject(fieldObj);
+                OBJECTHANDLE handle;
                 if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj))
                 {
-                    intptr_t ptr = (intptr_t)obj;
-                    memcpy(buffer, &ptr, sizeof(intptr_t));
+                    handle = (OBJECTHANDLE)obj;
+                    memcpy(buffer, &handle, sizeof(OBJECTHANDLE));
+                    result = true;
+                }
+                else if (!ignoreMovableObjects)
+                {
+                    handle = getJitHandleForObject(fieldObj);
```

Comment:
This will do the `IsInFrozenSegment` check again. Write it like this to avoid the redundant check?

```
if (ignoreMovableObjects)
{
    if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj))
    {
        handle = (CORINFO_OBJECT_HANDLE)obj;
        memcpy(buffer, &handle, sizeof(CORINFO_OBJECT_HANDLE));
        result = true;
    }
}
else
{
    handle = getJitHandleForObject(fieldObj);
    memcpy(buffer, &handle, sizeof(CORINFO_OBJECT_HANDLE));
}
```

Reply:
This does not appear to be addressed. We will still end up with two `IsInFrozenSegment` checks when `ignoreMovableObjects` is `false`.

Reply:
Ah, right, fixed, thanks!